diff --git a/subprojects/glib/gio/tests/meson.build b/subprojects/glib/gio/tests/meson.build
index b4a6492..509f6e1 100644
--- a/subprojects/glib/gio/tests/meson.build
+++ b/subprojects/glib/gio/tests/meson.build
@@ -928,7 +928,7 @@ if not meson.is_cross_build()
     test_resources_binary = custom_target('test_resources.o',
       input : test_gresource_binary,
       output : 'test_resources.o',
-      command : cc.cmd_array() + ['-Wl,-z,noexecstack', '-r', '-Wl,-b,binary',
+      command : cc.cmd_array() + ['-Wl,-z,noexecstack', '-r', '-Wl,-b,binary', '-nostdlib',
                                   '@INPUT@', '-o','@OUTPUT@'])
 
     # Rename symbol to match the one in the C file
